catalog/imaging/imaging-postprocess-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,556 @@
+    }
+  }
+
+  public void setMaxNitfSizeMB(int maxNitfSizeMB) {
+    this.maxNitfSizeMB = maxNitfSizeMB;
+  }
+
+  public void setCreateOverview(boolean createOverview) {
+    this.createOverview = createOverview;
+  }
+
+  public void setStoreOriginalImage(boolean storeOriginalImage) {
+    this.storeOriginalImage = storeOriginalImage;
+  }
+
+  private void handleProcessCreateItem(List<ProcessCreateItem> processCreateItems) {
+    List<ProcessCreateItem> createdItems = new ArrayList<>();
+    processCreateItems.stream().map(this::handleProcessCreateItem).forEach(createdItems::addAll);
+    processCreateItems.addAll(createdItems);
+  }
+
+  private List<ProcessCreateItem> handleProcessCreateItem(ProcessCreateItem processCreateItem) {
+    List<ProcessCreateItem> createdItems = new ArrayList<>();
+    Metacard metacard = processCreateItem.getMetacard();
+    ProcessResource processResource = processCreateItem.getProcessResource();
+
+    if (shouldProcess(processResource)) {
+      try (TemporaryFileBackedOutputStream fbos =
+          new TemporaryFileBackedOutputStream(DEFAULT_FILE_BACKED_OUTPUT_STREAM_THRESHOLD)) {
+
+        fbos.write(IOUtils.toByteArray(processResource.getInputStream()));
+        ByteSource byteSource = fbos.asByteSource();
+        BufferedImage renderedImage = renderImage(byteSource.openStream());
+
+        if (renderedImage != null) {
+          addThumbnailToMetacard(metacard, renderedImage);
+          processCreateItem.markMetacardAsModified();
+          if (createOverview) {
+            ProcessResource overviewProcessResource =
+                createOverviewResource(renderedImage, metacard);
+            createdItems.add(new ProcessCreateItemImpl(overviewProcessResource, metacard));
+          }
+
+          if (storeOriginalImage) {
+            ProcessResource originalImageProcessResource =
+                createOriginalImage(
+                    renderImageUsingOriginalDataModel(byteSource.openStream()), metacard);
+
+            createdItems.add(new ProcessCreateItemImpl(originalImageProcessResource, metacard));
+          }
+        }
+      } catch (IOException | NitfFormatException | RuntimeException e) {
+        LOGGER.debug(e.getMessage(), e);
+      } catch (InterruptedException e) {
+        LOGGER.error("Rendering failed for {}", processResource.getName(), e);
+      }
+    }
+    return createdItems;
+  }
+
+  public void handleProcessUpdateItem(List<ProcessUpdateItem> processUpdateItems) {
+    List<ProcessUpdateItem> updateItems = new ArrayList<>();
+    processUpdateItems.stream().map(this::handleProcessUpdateItem).forEach(updateItems::addAll);
+    processUpdateItems.addAll(updateItems);
+  }
+
+  public List<ProcessUpdateItem> handleProcessUpdateItem(ProcessUpdateItem processUpdateItem) {
+    List<ProcessUpdateItem> updatedItems = new ArrayList<>();
+    Metacard metacard = processUpdateItem.getMetacard();
+    Metacard originalMetacard = processUpdateItem.getOldMetacard();
```

Comment:
This method looks very similar to handleProcessCreateItem. Any way to refactor the common code out?